lib/ClangImporter/ClangImporter.cpp

```diff
@@ -3414,18 +3412,10 @@ ModuleDecl *ClangModuleUnit::getOverlayModule() const {
     // FIXME: Include proper source location.
     ModuleDecl *M = getParentModule();
     ASTContext &Ctx = M->getASTContext();
-    auto overlay = Ctx.getModuleByIdentifier(M->getName());
-    if (overlay == M) {
-      overlay = nullptr;
-    } else {
-      // FIXME: This bizarre and twisty invariant is due to nested
-      // re-entrancy in both clang module loading and overlay module loading.
-      auto *sharedModuleRef = Ctx.getLoadedModule(M->getName());
-      assert(!sharedModuleRef || sharedModuleRef == overlay ||
-             sharedModuleRef == M);
```

Comment:
I don't know this code, so I don't know where this assertion ended up. Don't think I've seen it yet.

Reply:
This assertion is saying "loading an overlay either fails (!), returns a clang module that is exactly our parent module, or returns a Swift module". Now that overlay loading has been rationalized, the condition would be "loading an overlay either fails or returns a Swift module", which is... not much of an invariant to assert about.